### PR TITLE
Migrate --upgrade_args to kubernetes_e2e.spy

### PR DESCRIPTION
**Pull request by fejta**
ref https://github.com/kubernetes/test-infra/issues/2829
/assign @krzyzacy @ixdy

### CODE REVIEW

**Comment by krzyzacy** [comment]
I feel like keep test args as some env might better help refactor the big json file?

**Comment by fejta** [comment]
I would like to migrate everything out of e2e-runner.sh so that everything is a flag we send to kubetest (so that I can delete e2e-runner.sh entirely).

Once we do that I'm happy to continue iterating on refactoring the config to make it more composable.

File: jenkins/e2e-image/e2e-runner.sh [hunk 1]
```diff
@@ -47,6 +47,14 @@ e2e_go_args=( \
...
 )
 
+if [[ "${FAIL_ON_GCP_RESOURCE_LEAK:-true}" == "true" ]]; then
+  case "${KUBERNETES_PROVIDER}" in
+    gce|gke)
+      e2e_go_args+=(--check-leaked-resources)
+      ;;
+  esac
+fi
+
```
Comment (krzyzacy) [inline jenkins/e2e-image/e2e-runner.sh]: rebase?

Comment (fejta) [inline jenkins/e2e-image/e2e-runner.sh]: Rebased

**Comment by fejta** [comment]
PTAL

**Pull request merged**